### PR TITLE
rgw: cls/user: set from_index for reset stats calls

### PR DESCRIPTION
**Pull request by theanalyst**
Fixes: https://tracker.ceph.com/issues/48327
Signed-off-by: Mykola Golub <mgolub@suse.com>
Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cls/user/cls_user.cc [hunk 1]
```diff
@@ -419,6 +419,9 @@ static int cls_user_reset_stats(cls_method_context_t hctx,
 	return -EIO;
       }
       add_header_stats(&header.stats, e);
+      if (!keys.empty()) {
+        from_index = keys.rbegin()->first;
+      }
```
Comment (cbodley) [inline src/cls/user/cls_user.cc:424]: shouldn't this go outside the `for (const auto& kv : keys) {}` loop?

Comment (theanalyst) [reply src/cls/user/cls_user.cc:424]: oops, C-c C-v error, fixing

**Comment by smithfarm** [comment]
```
CTest Failure
1 tests failed out of 211
The following tests FAILED: 	 13 - smoke.sh (Failed)
```

**Comment by smithfarm** [comment]
jenkins test make check

**Review by cbodley — APPROVED** [review]

**Comment by smithfarm** [comment]
@cbodley @theanalyst Would be great to get this into an upcoming rgw suite. Do you know who I might ~pester~ask nicely about that?

**Comment by harishmunjulur** [comment]
http://pulpito.front.sepia.ceph.com:80/hmunjulu-2021-01-07_19:17:35-rgw-wip-hmunjulu01-testing-2021-01-07-0741-distro-basic-smithi/

https://trello.com/c/O4OGYtgH

**Pull request merged**